pkg/logs/config/integration_config.go

```diff
@@ -213,6 +213,10 @@ func ValidateProcessingRules(rules []LogsProcessingRule) error {
 		if rule.Name == "" {
 			return fmt.Errorf("LogsAgent misconfigured: all log processing rules need a name")
 		}
+		_, err := regexp.Compile(rule.Pattern)
```

Comment:
I would rather change `CompileProcessingRules` to return an error and use `regexp.Compile`. As @tmichelet it's better to avoid compiling the regex twice since it's an expensive operation.

Reply:
(for me it's not that the operation is expensive (it runs only at startup anyway), it's more that it's error prone to leave as is)

Reply:
@ajacquemot I can add to `CompileProcessingRules`. Then, compiling twice would be needed only for `case MultiLine`.